manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/persistence/dao/AbstractConnectorDAOTest.java

```diff
@@ -0,0 +1,145 @@
+        assertThat(retrieved.getTopicName()).isEqualTo(TestConstants.DEFAULT_CONNECTOR_TOPIC_NAME);
+        assertThat(retrieved.getTopicName()).isEqualTo(TestConstants.DEFAULT_CONNECTOR_TOPIC_NAME);
+        assertThat(retrieved.getOperation()).isNotNull();
+        assertThat(retrieved.getOperation().getType()).isNotNull();
+        assertThat(retrieved.getOperation().getRequestedAt()).isNotNull();
+        assertThat(retrieved.getConditions()).isNull();
+        assertThat(retrieved.getError()).isNull();
+    }
+
+    @Test
+    @Transactional
+    public void testStoreConnectorWithConditions() {
+        Bridge bridge = createBridge();
+        bridgeDAO.persist(bridge);
+
+        Connector connector = Fixtures.createReadyConnector(bridge, getConnectorType());
+        getConnectorDAO().persist(connector);
+
+        Connector retrieved = getConnectorDAO().findByIdWithConditions(connector.getId());
+
+        assertThat(retrieved.getId()).isEqualTo(connector.getId());
+        assertThat(retrieved.getConditions()).hasSize(2);
+    }
+
+    @Test
+    @Transactional
+    public void findByShardIdToDeployOrDeleteWhenControlPlaneIsComplete() {
+        Bridge b = createBridge();
+        Connector c = createConnector(b, getConnectorType());
+        bridgeDAO.persist(b);
+
+        Condition condition1 = createCondition();
+        Condition condition2 = createCondition();
+        Condition condition3 = createCondition();
+        condition3.setComponent(ComponentType.SHARD);
+        condition3.setStatus(ConditionStatus.UNKNOWN);
+
+        c.setConditions(List.of(condition1, condition2, condition3));
+        getConnectorDAO().persist(c);
+
+        List<Connector> connectors = getConnectorDAO().findByShardIdToDeployOrDelete(b.getShardId());
+        assertThat(connectors).isNotNull();
+        assertThat(connectors).hasSize(1);
+        assertThat(connectors.get(0).getId()).isEqualTo(c.getId());
+    }
+
+    @Test
+    @Transactional
+    public void findByShardIdToDeployOrDeleteWhenControlPlaneIsIncomplete() {
+        Bridge b = createBridge();
+        Connector c = createConnector(b, getConnectorType());
+        bridgeDAO.persist(b);
+
+        Condition condition1 = createCondition();
+        Condition condition2 = createCondition();
+        condition2.setStatus(ConditionStatus.UNKNOWN);
+
+        Condition condition3 = createCondition();
+        condition3.setComponent(ComponentType.SHARD);
+        condition3.setStatus(ConditionStatus.UNKNOWN);
+
+        c.setConditions(List.of(condition1, condition2, condition3));
+        getConnectorDAO().persist(c);
+
+        List<Connector> processors = getConnectorDAO().findByShardIdToDeployOrDelete(b.getShardId());
+        assertThat(processors).isNotNull();
+        assertThat(processors).isEmpty();
+    }
+
+    @Test
+    @Transactional
+    public void testTypeDiscriminator() {
```

Comment:
What does this actually test!?!

It creates different types of `Connector`'s and then asserts it cannot be found.

Reply:
As we use the same table for both sink and sources, it tests that the queries are actually filtering by type. I can remove if you think it is not relevant

Reply:
Oh, OK. I see the relevance of the cryptic "type inversion" on L#130.

Create a Connector for the "other type" and it should not be found. Makes sense. Just difficult to see.

Reply:
You're right, I'll add a comment to make it more clear :+1: